contracts/src/v0.8/functions/dev/1_0_0/FunctionsBilling.sol

```diff
@@ -0,0 +1,421 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.6;
+
+import {Route, ITypeAndVersion} from "./Route.sol";
+import {IFunctionsRouter} from "./interfaces/IFunctionsRouter.sol";
+import {IFunctionsSubscriptions} from "./interfaces/IFunctionsSubscriptions.sol";
+import {LinkTokenInterface} from "../../../interfaces/LinkTokenInterface.sol";
+import {AggregatorV3Interface} from "../../../interfaces/AggregatorV3Interface.sol";
+import {IFunctionsBilling} from "./interfaces/IFunctionsBilling.sol";
+import {IFunctionsClient} from "./interfaces/IFunctionsClient.sol";
+import {ERC677ReceiverInterface} from "../../../interfaces/ERC677ReceiverInterface.sol";
+import {IAuthorizedOriginReceiver} from "./accessControl/interfaces/IAuthorizedOriginReceiver.sol";
+import {SafeCast} from "../../../shared/vendor/openzeppelin-solidity/v.4.8.0/contracts/utils/SafeCast.sol";
+
+/**
+ * @title Functions Billing contract
+ * @notice Contract that calculates payment from users to the nodes of the Decentralized Oracle Network (DON).
+ * @dev THIS CONTRACT HAS NOT GONE THROUGH ANY SECURITY REVIEW. DO NOT USE IN PROD.
+ */
+abstract contract FunctionsBilling is Route, IFunctionsBilling {
+  AggregatorV3Interface private LINK_TO_NATIVE_FEED;
+
+  // ================================================================
+  // |                  Request Commitment state                    |
+  // ================================================================
+  struct Commitment {
+    uint64 subscriptionId;
+    address client;
+    uint32 gasLimit;
+    uint256 gasPrice;
```

Comment:
Can we rename to callbackGasLimit and callbackGasPrice?

Reply:
changed to `callbackGasLimit` and `expectedGasPrice`